extensions/emmet/src/abbreviationActions.ts

```diff
@@ -617,11 +617,19 @@ function expandAbbr(input: ExpandAbbreviationInput): string {
 				if (wrappingNode && inlineElements.indexOf(wrappingNode.name) === -1) {
 					wrappingNode.value = '\n\t' + wrappingNode.value + '\n';
 				}
+
+				if (wrappingNode.name === 'a') {
+					isAnchor = true;
+				}
 			}
 			expandedText = helper.expandAbbreviation(parsedAbbr, expandOptions);
 			// All $anyword would have been escaped by the emmet helper.
 			// Remove the escaping backslash from $TM_SELECTED_TEXT so that VS Code Snippet controller can treat it as a variable
 			expandedText = expandedText.replace('\\$TM_SELECTED_TEXT', '$TM_SELECTED_TEXT');
+
+			if (isAnchor) {
+				expandedText = expandedText.replace('${0}', '$TM_SELECTED_TEXT');
```

Comment:
The problems with this approach are
- We are assuming that `${0}` will always correspond to the `href` attribute. Which may not be the case, if the user is using custom snippets.
- In case of `Wrap Individual Lines with Abbreviation`, there is no `$TM_SELECTED_TEXT`
- The replacement would happen even if the wrapped text is not a url or an email

Reply:
Gotcha, thanks for the feedback.
I was checking and the bug does not happen if you use "Wrap Individual Lines with Abbreviation"